bindings/wasm/src/did/wasm_document.rs

```diff
@@ -87,10 +98,84 @@ impl WasmDocument {
     WasmDID(self.0.id().clone())
   }
 
+  /// Returns a list of the document controllers.
+  #[wasm_bindgen(js_name = controller)]
+  pub fn controller(&self) -> DIDArray {
+    match self.0.controller() {
+      Some(controllers) => controllers
+        .iter()
+        .cloned()
+        .map(WasmDID::from)
+        .map(JsValue::from)
+        .collect::<js_sys::Array>()
+        .unchecked_into::<DIDArray>(),
+      None => js_sys::Array::new().unchecked_into::<DIDArray>(),
+    }
+  }
+
+  /// Returns a set of the document's `alsoKnownAs`.
+  #[wasm_bindgen(js_name = alsoKnownAs)]
+  pub fn also_known_as(&self) -> DIDArray {
+    self
+      .0
+      .also_known_as()
+      .iter()
+      .cloned()
+      .map(|url| url.to_string())
+      .map(JsValue::from)
+      .collect::<js_sys::Array>()
+      .unchecked_into::<DIDArray>()
+  }
+
+  /// Adds a custom property to the DID Document.
+  /// If the Document already has this key, the value will be overwritten.
+  /// If the value is set to `null`, the property will be removed.
+  ///
+  /// ### WARNING
+  /// This method can overwrite existing properties like `id`.
+  #[wasm_bindgen(js_name = setPropertyUnchecked)]
+  pub fn set_property(&mut self, key: String, value: &JsValue) -> Result<()> {
+    let value: Option<serde_json::Value> = value.into_serde().wasm_result()?;
+    match value {
+      Some(value) => {
+        self.0.properties_mut().insert(key, value);
+      }
+      None => {
+        self.0.properties_mut().remove(&key);
+      }
+    }
+    Ok(())
+  }
+
+  /// Returns the custom DID Document properties.
+  #[wasm_bindgen(js_name = properties)]
+  pub fn properties(&mut self) -> Result<JSMap> {
+    let properties = self.0.properties();
+    let properties_map = js_sys::Map::new();
+    for (key, value) in properties.iter() {
```

Comment:
Nit: remove unnecessary variable.

```suggestion
    let properties_map = js_sys::Map::new();
    for (key, value) in self.0.properties().iter() {
```